core/src/avm2/globals/flash/text/text_field.rs

```diff
@@ -1325,18 +1325,41 @@ pub fn set_mouse_wheel_enabled<'gc>(
 
 pub fn get_restrict<'gc>(
     activation: &mut Activation<'_, 'gc>,
-    _this: Object<'gc>,
+    this: Object<'gc>,
     _args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    avm2_stub_getter!(activation, "flash.text.TextField", "restrict");
-    Ok(Value::Null)
+    if let Some(this) = this
+        .as_display_object()
+        .and_then(|this| this.as_edit_text())
+    {
+        return match this.restrict() {
+            Some(value) => Ok(AvmString::new(activation.context.gc_context, value).into()),
+            None => Ok(Value::Null),
+        };
+    }
+
+    Ok(Value::Undefined)
 }
 
 pub fn set_restrict<'gc>(
     activation: &mut Activation<'_, 'gc>,
-    _this: Object<'gc>,
-    _args: &[Value<'gc>],
+    this: Object<'gc>,
+    args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    avm2_stub_setter!(activation, "flash.text.TextField", "restrict");
+    if let Some(this) = this
+        .as_display_object()
+        .and_then(|this| this.as_edit_text())
+    {
+        let value = args.get(0).cloned().unwrap_or(Value::Undefined);
```

Comment:
We have `get_object` and `try_get_object` in `ParametersExt`, perhaps that could be used here?

Reply:
I replaced it with `try_get_string`, should look better now!